lib/fsm/logger.go

```diff
@@ -42,31 +45,37 @@ type Logger struct {
 
 // Debug logs a debug message
 func (l *Logger) Debug(args ...interface{}) {
+	initQueue()
 	l.FieldLogger.Debug(args...)
-	err := l.Operator.CreateLogEntry(l.Key, l.makeLogEntry(
-		fmt.Sprint(args...), "debug"))
-	if err != nil {
-		l.FieldLogger.Error(trace.DebugReport(err))
+
+	select {
+	case logEntryC <- l.makeLogEntry(fmt.Sprint(args...), "debug"):
+	default:
+		l.FieldLogger.Debug("operation logger dropped message: ", fmt.Sprint(args...))
 	}
 }
 
 // Info logs an info message
 func (l *Logger) Info(args ...interface{}) {
+	initQueue()
 	l.FieldLogger.Info(args...)
-	err := l.Operator.CreateLogEntry(l.Key, l.makeLogEntry(
-		fmt.Sprint(args...), "info"))
-	if err != nil {
-		l.FieldLogger.Error(trace.DebugReport(err))
+
+	select {
+	case logEntryC <- l.makeLogEntry(fmt.Sprint(args...), "info"):
+	default:
+		l.FieldLogger.Info("operation logger dropped message: ", fmt.Sprint(args...))
```

Comment:
The other similar messages seem to log at Debug while this one is Info for some reason.

Reply:
good catch